quickwit/quickwit-cluster/src/cluster.rs

```diff
@@ -134,6 +134,11 @@ impl Cluster {
             failure_detector_config,
             gossip_interval,
             marked_for_deletion_grace_period: MARKED_FOR_DELETION_GRACE_PERIOD,
+            liveness_predicate: Some(Box::new(|node_state: &NodeState| {
+                node_state.contains_key(GRPC_ADVERTISE_ADDR_KEY) &&
+                node_state.contains_key(ENABLED_SERVICES_KEY) &&
+                node_state.get(READINESS_KEY) == Some(READINESS_VALUE_READY)
```

Comment:
```suggestion
                node_state.contains_key(ENABLED_SERVICES_KEY)
```
There's already some logic in place to distinguish ready/not ready live nodes. When a node is flapping between the two states, we output some logs that are useful for debugging. We just need the first two for correctness.